app/services/builder_base_service.rb

```diff
@@ -0,0 +1,43 @@
+class BuilderBaseService
+  def self.build
+    new
+  end
+
+  def self.call(*args, **kwargs)
+    inst = build
+    result = nil
+    errors = []
+
+    # Establish basic interface
+    unless inst.respond_to?(:call)
+      raise NotImplementedError.new("Contract violation, Builder Service Children must implement call")
+    end
+
+    # Error handling by default
+    begin
+      result = inst.call(*args, **kwargs)
+    rescue => e
+      errors.append(e)
+    end
+
+    OpenStruct.new(success?: errors.empty?, errors: errors, result: result)
+  end
+
+  def call(*args, **kwargs)
```

Comment:
@tsmartt This is an example of `#2` on my comment above.  Primary child class api is `run` method (or whatever, just for example).

Primary usage is `Service.build.call(*args, **kwargs)` and developer implements required `run` interface and optionally overrides `Class.build`